itest/lnd_sweep_test.go

```diff
@@ -1244,3 +1245,289 @@ func createSimpleNetwork(ht *lntest.HarnessTest, nodeCfg []string,
+		txns := ht.Miner.GetNumTxsFromMempool(2)
+
+		// Find the sweeping tx - assume it's the first item, if it has
+		// the same txid as the parent tx, use the second item.
+		sweepTx := txns[0]
+		if sweepTx.TxHash() == tx.TxHash() {
+			sweepTx = txns[1]
+		}
+
+		return pendingSweep.SatPerVbyte, sweepTx
+	}
+
+	// First bump request - we'll sepcify nothing except `Immediate` to let
+	// the sweeper handles the fee, and we expect a fee func that has,
+	// - starting fee rate: 1 sat/vbyte (min relay fee rate).
+	// - deadline: 1008 (default deadline).
+	// - budget: 50% of the input value.
+	bumpFeeReq := &walletrpc.BumpFeeRequest{
+		Outpoint: op,
+		// We use a force param to create the sweeping tx immediately.
+		Immediate: true,
+	}
+	alice.RPC.BumpFee(bumpFeeReq)
+
+	// Since the request doesn't specify a deadline, we expect the default
+	// deadline to be used.
+	_, currentHeight := ht.Miner.GetBestBlock()
+	deadline := uint32(currentHeight + sweep.DefaultDeadlineDelta)
+
+	// Assert the pending sweep is created with the expected values:
+	// - broadcast attempts: 1.
+	// - starting fee rate: 1 sat/vbyte (min relay fee rate).
+	// - deadline: 1008 (default deadline).
+	// - budget: 50% of the input value.
+	feerate1, sweepTx1 := assertPendingSweepResp(
+		1, uint64(value/2), deadline, 0,
+	)
+
+	// Since the request doesn't specify a starting fee rate, we expect the
+	// min relay fee rate is used as the current fee rate.
+	require.Equal(ht, startFeeRate, feerate1, "current fee rate not match")
+
+	// testFeeRate sepcifies a starting fee rate in sat/vbyte.
+	const testFeeRate = uint64(100)
+
+	// Second bump request - we will specify the fee rate and expect a fee
+	// func that has,
+	// - starting fee rate: 100 sat/vbyte.
+	// - deadline: 1008 (default deadline).
+	// - budget: 50% of the input value.
+	bumpFeeReq = &walletrpc.BumpFeeRequest{
+		Outpoint: op,
+		// We use a force param to create the sweeping tx immediately.
+		Immediate:   true,
+		SatPerVbyte: testFeeRate,
+	}
+	alice.RPC.BumpFee(bumpFeeReq)
+
+	// Assert the pending sweep is created with the expected values:
+	// - broadcast attempts: 2.
+	// - starting fee rate: 100 sat/vbyte.
+	// - deadline: 1008 (default deadline).
+	// - budget: 50% of the input value.
+	feerate2, sweepTx2 := assertPendingSweepResp(
+		2, uint64(value/2), deadline, testFeeRate,
+	)
+
+	// We expect the requested starting fee rate to be the current fee
+	// rate.
+	require.Equalf(ht, testFeeRate, feerate2, "current fee rate not "+
+		"match: want: %d, got: %d", testFeeRate, feerate2)
+
+	// Alice's old sweeping tx should be replaced.
+	ht.Miner.AssertTxNotInMempool(sweepTx1.TxHash())
+
+	// testBudget specifies a budget in sats.
+	testBudget := uint64(float64(value) * 0.1)
+
+	// Third bump request - we will specify the budget and expect a fee
+	// func that has,
+	// - starting fee rate: 100 sat/vbyte, stays unchanged.
+	// - deadline: 1008 (default deadline).
+	// - budget: 10% of the input value.
```

Comment:
Are we allowed to lower the budget using BumpFee?

Reply:
With the end goal being to reduce the fee rate of the broadcasted transaction? I don't think there's a practical way of doing so, since replacement is based generally on continuing to pay more fees.

Reply:
Right that's why I'm asking. This test case appears to lower the budget in subsequent bump attempts and that struck me as odd.

Reply:
yeah you are allowed to lower it - but the resulting replacement tx may not succeed. It really depends on what the current fee rate is, as the budget only specifies the ending fee rate of the fee func.